js/anxpro.js

```diff
@@ -80,6 +122,327 @@ module.exports = class anxpro extends Exchange {
+        //
+        // deposit:
+        //    {
+        //     "transactionClass": "COIN",
+        //     "uuid": "eb65576f-c1a8-423c-8e2f-fa50109b2eab",
+        //     "userUuid": "82027ee9-cb59-4f29-80d6-f7e793f39ad4",
+        //     "amount": 3.99287184,
+        //     "fee": 0,
+        //     "balanceBefore": 8.39666034,
+        //     "balanceAfter": 12.38953218,
+        //     "ccy": "ETH",
+        //     "transactionState": "PROCESSED",
+        //     "transactionType": "DEPOSIT",
+        //     "received": "1529420056000",
+        //     "processed": "1529420766000",
+        //     "timestampMillis": "1557442743854",
+        //     "displayTitle": "Coin Deposit",
+        //     "displayDescription": "Deposit to: 0xf123aa44fadea913a7da99cc2ee202db684ce0e3",
+        //     "coinTransactionId": "0x33a3e5ea7c034dc5324a88aa313962df0a5d571ab4bcc3cb00b876b1bdfc54f7",
+        //     "coinConfirmations": 51,
+        //     "coinConfirmationsRequired": 45,
+        //     "subAccount": {"uuid": "aba1de05-c7c6-49d7-84ab-a6aca0e827b6", "name": "DEFAULT"}
+        //    }
+        const timestamp = this.safeInteger (transaction, 'received');
+        const updated = this.safeInteger (transaction, 'processed');
+        const transactionType = this.safeString (transaction, 'transactionType');
+        let type = undefined;
+        let amount = this.safeFloat (transaction, 'amount');
+        let address = this.safeString (transaction, 'coinAddress');
+        let tag = undefined;
+        if (transactionType === 'WITHDRAWAL') {
+            type = 'withdrawal';
+            amount = -amount;
+            if (address) {
+                //  xrp: "coinAddress": "rw2ciyaNshpHe7bCHo4bRWq6pqqynnWKQg?dt=3750180345",
+                if (address.indexOf ('?dt=') >= 0) {
+                    const parts = address.split ('?dt=');
+                    address = parts[0];
+                    tag = parts[1];
+                }
+            }
+        } else if (transactionType === 'DEPOSIT') {
+            if (!address) {
+                const displayDescription = this.safeString (transaction, 'displayDescription');
+                const addressText = displayDescription.replace ('Deposit to: ', '');
+                if (addressText.length > 0) {
+                    //  eth: "displayDescription": "Deposit to: 0xf123aa44fadea913a7da99cc2ee202db684ce0e3",
+                    //  xrp: "displayDescription": "Deposit to: rUjxty1WWLwX1evhKf3C2XNZDMcXEZ9ToJ?dt=504562345",
+                    if (addressText.indexOf ('?dt=') >= 0) {
+                        const parts = addressText.split ('?dt=');
+                        address = parts[0];
+                        tag = parts[1];
+                    } else {
+                        address = addressText;
+                    }
+                }
+            }
+            type = 'deposit';
+        }
+        const ccy = this.safeString (transaction, 'ccy');
```

Comment:
We need to rework the five lines 264-268, because this particular code breaks the common currency and currency code unification rules, so, that section has to be:

```JavaScript
const currencyId = this.safeString (transaction, 'ccy'); // exchange-specific code
let code = undefined; // unified code
if (currencyId in this.currencies_by_id) {
    // if there's a matching preloaded currency, take the code by id indirectly
    code = this.currencies_by_id[currencyId']['code'];
} else {
    // otherwise convert the string to a unified code explicitly
    code = this.commonCurrencyCode (currencyId); 
}
```

or

```JavaScript
const currencyId = this.safeString (transaction, 'ccy'); // exchange-specific code
const code = this.commonCurrencyCode (currencyId); // unified code
```